pkg/providers/vsphere/controlplane.go

```diff
@@ -52,6 +58,77 @@ func (b *ControlPlaneBuilder) BuildFromParsed(lookup yamlutil.ObjectLookup) erro
 	return nil
 }
 
+// ControlPlaneSpec builds a vsphere ControlPlane definition based on an eks-a cluster spec.
+func ControlPlaneSpec(ctx context.Context, logger logr.Logger, client kubernetes.Client, spec *cluster.Spec) (*ControlPlane, error) {
+	templateBuilder := NewVsphereTemplateBuilder(time.Now, true)
+
+	controlPlaneYaml, err := templateBuilder.GenerateCAPISpecControlPlane(
+		spec,
+		func(values map[string]interface{}) {
+			values["controlPlaneTemplateName"] = clusterapi.ControlPlaneMachineTemplateName(spec)
+			values["etcdTemplateName"] = clusterapi.EtcdMachineTemplateName(spec)
```

Comment:
Can we avoid passing overrides? I would like for the template builder to generate everything that is needed so we have the logic only in one place.

Reply:
I see I see. That seems quite a bit of work for little benefit. Let's chat about this offline.